aiida/manage/database/delete/nodes.py

```diff
@@ -44,18 +44,27 @@ def delete_nodes(pks,
         The verbosity levels, 0 prints nothing, 1 prints just sums and total, 2 prints individual nodes.
     """
     # pylint: disable=too-many-arguments,too-many-branches,too-many-locals,too-many-statements
-    from aiida.orm.querybuilder import QueryBuilder
+    from aiida.common import exceptions
     from aiida.common.links import LinkType
     from aiida.orm.node import Node
     from aiida.orm.node import ProcessNode
     from aiida.orm.node.data import Data
+    from aiida.orm.querybuilder import QueryBuilder
     from aiida.orm import load_node
     from aiida.backends.utils import delete_nodes_and_connections
 
     user_email = User.objects.get_default().email
 
-    if not pks:
-        # If I was passed an empty list, I don't to anything
+    starting_pks = []
+    for pk in pks:
+        try:
+            load_node(pk)
+        except exceptions.NotExistent:
+            print('warning: node with pk<{}> does not exist, skipping'.format(pk))
```

Comment:
Shouldn't we use click.echo?

Reply:
Didn't touch that part, but might as well, you are right